### PR TITLE
feat(backup): Validation steps for relocation tasks

### PR DESCRIPTION
**Pull request by azaslavsky**
These steps start, poll, and resolve a single Google CloudBuild validation run.

Issue: getsentry/team-ospo#203

### CODE REVIEW

File: src/sentry/tasks/relocation.py [hunk 1]
```diff
@@ -456,18 +492,143 @@ def preprocessing_complete(uuid: str) -> None:
...
+            # result.
+            if relocation_validation.status < status.value:
+                relocation_validation.status = status.value
+                relocation_validation_attempt.save()
+            return fail_relocation(
+                relocation, task, "Validation could not be completed. Please contact support."
+            )
+
+        # All remaining statuses are final, so we can update the owning `RelocationValidation` now.
+        assert status in {ValidationStatus.INVALID, ValidationStatus.VALID}
```
Comment (hubertdeng123) [inline src/sentry/tasks/relocation.py:597]: Would it be better to throw a more specific error if the status is unexpected here? 

`throw new Error('Expected validation status of INVALID or VALID, received ${status}`

**Review by hubertdeng123 — APPROVED** [review]

File: src/sentry/tasks/relocation.py (hunk 1 above)
Comment (azaslavsky) [reply src/sentry/tasks/relocation.py:597]: I think for this case, where I am trying to state an invariant, `assert` is the more idiomatic way to go? Throwing an exception implies that there is a valid program state that could cause this to be untrue, which is not the case here. The main reason for the `assert` is: 1. Clarity for readers ("you may assume for the rest of this method that that only these statuses are still possible, and 2. To create an invariant that guards against statuses being added/removed.

Comment (hubertdeng123) [reply src/sentry/tasks/relocation.py:597]: I think you're right, seeing something similar in other areas of code too

File: src/sentry/tasks/relocation.py [hunk 2]
```diff
@@ -477,5 +638,244 @@
...
+        fail_relocation(relocation, OrderedTask.VALIDATING_START, ERR_VALIDATING_MAX_RUNS)
+        return
+
+    with retry_task_or_fail_relocation(
+        relocation, OrderedTask.VALIDATING_START, attempts_left, ERR_VALIDATING_INTERNAL
+    ):
+        cb_client = CloudBuildClient()
+
+        def camel_to_snake_keep_underscores(value):
+            match = re.search(r"(_+)$", value)
```
[the remark this answers is not in the document]
Comment (azaslavsky) [reply src/sentry/tasks/relocation.py:665]: This is not user-provided data, only data in a template, so this is fine.

**Pull request merged**